app/src/main/java/de/danoeh/antennapod/activity/MainActivity.java

```diff
@@ -144,13 +146,23 @@ public void onCreate(Bundle savedInstanceState) {
 
         checkFirstLaunch();
         PreferenceUpgrader.checkUpgrades(this);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            this.registerLollipopNetworkListener();
```

Comment:
Hmm, I think it is a bit weird to do this from the UI. What if the download service starts automatically in the background and the user does not launch the UI? I would do that in DownloadService.

Should we unregister as well, when the downloads are finished? Otherwise we might get registered multiple times. (Something like that caused issues in the playback service recently)